app_dart/lib/src/request_handlers/refresh_chromebot_status.dart

```diff
@@ -49,25 +50,26 @@ class RefreshChromebotStatus extends ApiRequestHandler<Body> {
     );
 
     for (LuciBuilder builder in luciTasks.keys) {
-      await config.db.withTransaction<void>((Transaction transaction) async {
-        try {
-          await for (FullTask task
-              in datastore.queryRecentTasks(taskName: builder.taskName)) {
-            for (LuciTask luciTask in luciTasks[builder]) {
-              if (luciTask.commitSha == task.commit.sha) {
-                final Task update = task.task;
-                update.status = luciTask.status;
-                transaction.queueMutations(inserts: <Task>[update]);
-                // Stop updating task whenever we find the latest status of the same commit.
-                break;
+      await runTransactionWithRetries(() async {
+        await config.db.withTransaction<void>((Transaction transaction) async {
+          try {
+            await for (FullTask task
+                in datastore.queryRecentTasks(taskName: builder.taskName)) {
+              for (LuciTask luciTask in luciTasks[builder]) {
+                if (luciTask.commitSha == task.commit.sha) {
+                  final Task update = task.task;
+                  update.status = luciTask.status;
+                  transaction.queueMutations(inserts: <Task>[update]);
+                  // Stop updating task whenever we find the latest status of the same commit.
+                  break;
+                }
               }
             }
+            await transaction.commit();
+          } catch (error) {
+            rethrow;
           }
-          await transaction.commit();
-        } catch (error) {
-          await transaction.rollback();
-          rethrow;
-        }
+        });
       });
     }
```

Comment:
@godofredoc I'm talking about reducing nesting in general, for example, this change has 6 nested scopes, which made it a little difficult to read. Also see go/tott/455.

Reply:
I see what you mean, thanks for the explanation. Those several layers of nested scopes are  everywhere where the backend is using transactions.

Added a new bug for the cleanup as it will require to update most of the handlers:
 
https://github.com/flutter/flutter/issues/52436